power10dan/tldrResearchPaper
# Python update

## What this branch does
This branch migrates django from python 2.7 to 3.6

## How to test this
All you need to do is make a new virtual environment like this:
```
SE/tldrResearchPaper | mkvirtualenv upgradedDjango
Using base prefix '/usr'
New python executable in /home/doyougnu/.virtualenvs/upgradedDjango/bin/python
Installing setuptools, pip, wheel...done.
virtualenvwrapper.user_scripts creating /home/doyougnu/.virtualenvs/upgradedDjango/bin/predeactivate
virtualenvwrapper.user_scripts creating /home/doyougnu/.virtualenvs/upgradedDjango/bin/postdeactivate
virtualenvwrapper.user_scripts creating /home/doyougnu/.virtualenvs/upgradedDjango/bin/preactivate
virtualenvwrapper.user_scripts creating /home/doyougnu/.virtualenvs/upgradedDjango/bin/postactivate
virtualenvwrapper.user_scripts creating /home/doyougnu/.virtualenvs/upgradedDjango/bin/get_env_details
(upgradedDjango) SE/tldrResearchPaper |
```

Then migrate to requirements and install the file using pip:

```
(upgradedDjango) SE/tldrResearchPaper | pip install -r uploads/requirements/requirements_dev.txt 
Collecting Django==1.11.5 (from -r uploads/requirements/requirements_dev.txt (line 5))
  Using cached Django-1.11.5-py2.py3-none-any.whl
Collecting djangorestframework==3.6.4 (from -r uploads/requirements/requirements_dev.txt (line 6))
  Using cached djangorestframework-3.6.4-py2.py3-none-any.whl
Collecting psycopg2==2.7.3.1 (from -r uploads/requirements/requirements_dev.txt (line 7))
  Using cached psycopg2-2.7.3.1-cp36-cp36m-manylinux1_x86_64.whl
...
...
...
Successfully installed Django-1.11.5 Pygments-2.2.0 attrs-17.2.0 autoflake-0.7 certifi-2017.7.27.1 chardet-3.0.4 configparser-3.5.0 coverage-4.4.1 defusedxml-0.5.0 django-allauth-0.33.0 django-cors-headers-2.1.0 django-extensions-1.9.1 django-rest-auth-0.9.2 djangorestframework-3.6.4 enum34-1.1.6 flake8-3.4.1 httpie-0.9.9 hypothesis-3.32.1 idna-2.6 jedi-0.11.0 json-rpc-1.10.3 mccabe-0.6.1 nose-1.3.7 oauthlib-2.0.5 parso-0.1.0 psycopg2-2.7.3.1 py4j-0.10.6 pycodestyle-2.3.1 pyflakes-1.5.0 python-openid-2.2.5 python3-openid-3.1.0 pytz-2017.2 requests-2.18.4 requests-oauthlib-0.8.0 service-factory-0.1.5 six-1.11.0 urllib3-1.22
(upgradedDjango) SE/tldrResearchPaper |
```

Make sure to migrate:
```
(upgradedDjango) SE/tldrResearchPaper | python manage.py migrate
Operations to perform:
  Apply all migrations: account, admin, auth, authtoken, contenttypes, sessions, sites, uploads
Running migrations:
  Applying account.0001_initial... OK
  Applying account.0002_email_max_length... OK
  Applying authtoken.0001_initial... OK
  Applying authtoken.0002_auto_20160226_1747... OK
  Applying sites.0001_initial... OK
  Applying sites.0002_alter_domain_unique... OK
  Applying uploads.0001_initial... OK
  Applying uploads.0002_auto_20171014_0425... OK
```

and now run the server:
```
(upgradedDjango) SE/tldrResearchPaper | python manage.py runserver              PythonUpdate | 10:43 AM
Performing system checks...

System check identified no issues (0 silenced).
October 26, 2017 - 17:44:28
Django version 1.11.5, using settings 'uploads.settings.settings_dev'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
```

**IF THIS IS NOT REVIEWED IN 24 HOURS I'M MERGING**
